test/src/features/chat/data/model/chat_model_test.dart

```
@@ -0,0 +1,121 @@
+        expect(json['avatarUrl'], 'https://example.com/avatar.jpg');
+        expect(json['lastMessage'], 'Hello, how are you?');
+        expect(json['createdTime'], expectedLocalTime);
+        expect(json['isArchived'], false);
+      });
+    });
+
+    // 边缘情况测试组
+    group('Edge Cases', () {
+      test('fromJson should handle missing fields with default values', () {
+        // 测试处理缺失字段时的默认值
+        final json = {
+          'id': 'abc123',
+        };
+
+        final model = ContactModel.fromJson(json);
+
+        // 验证默认值是否正确设置
+        expect(model.id, 'abc123');
+        expect(model.userName, 'Unknown User');
+        expect(model.avatarUrl, '');
+        expect(model.lastMessage, '');
+        expect(model.createdTime, null);
+        expect(model.isArchived, false);
+      });
+
+      test('fromJson should generate new UUID when id is missing', () {
+        // 测试缺失ID时是否自动生成UUID
+        final json = {
+          'userName': 'John Doe',
+        };
+
+        final model = ContactModel.fromJson(json);
+
+        // 验证生成的UUID是否有效
+        expect(model.id, isNotEmpty);
+        expect(model.id.length, 36);
+      });
+    });
+
+    // 日期时间处理测试组
+    group('DateTime Handling', () {
+      test('fromJson should handle invalid createdTime format', () {
+        // 测试处理无效的日期格式
+        final json = {
+          'id': 'abc123',
+          'createdTime': 'invalid-date-format',
+        };
+
+        final model = ContactModel.fromJson(json);
+
+        // 验证无效日期是否被正确处理为null
+        expect(model.createdTime, null);
+      });
+
+      test('toJson should handle null createdTime', () {
+        // 测试处理null日期时间
+        final model = ContactModel(
+          id: 'abc123',
+          userName: 'John Doe',
+          avatarUrl: '',
+          lastMessage: '',
+          createdTime: null,
+          isArchived: false,
+        );
+
+        final json = model.toJson();
+
+        // 验证null日期时间是否被转换为空字符串
+        expect(json['createdTime'], '');
```

Same as the previous test, the implementation throws `FormatException` instead of converting `createdTime` to empty string.